File: Aptitute/do while loop/HCF.java
public class HCF {
    public static void main(String[] args) {
        int a = 300;
        int b = 20;
        int temp = 0;
        do{
            if(a > b){
                temp = b;
                for(int i = temp; i > 0 ;i--){
                    if(a%i ==0 && b %i ==0){
                        System.out.println(i);
                        break;
                    }
                }
            }
            else{
                temp = a;
                for(int i = temp; i > 0 ;i--){
                    if(a%i ==0 && b %i ==0){
                        System.out.println(i);
                        break;
                    }
                }
            }
        }while(false);
    }
}

File: c course/New folder (2)/hcf.c
#include<stdio.h>
int main(){
    int min,a,b,i,hcf;
    printf("enter the value of a and b:");
    scanf("%d %d",&a,&b);
    min =(a<b)?a:b;
    for(i=1;i<=min;i++){
        if(a%i==0 && b%i==0){
            hcf =i;
        }
    }
    printf("hcf of %d %d is%d:",a,b,hcf);
    return 0;
}